src/gui/qgsqueryresultwidget.cpp

```diff
@@ -172,7 +172,7 @@ void QgsQueryResultWidget::executeQuery()
   cancelRunningQuery();
   if ( mConnection )
   {
-    const QString sql { mSqlEditor->text( ) };
+    const QString sql = mSqlEditor->selectedText().length() > 0 ? mSqlEditor->selectedText() : mSqlEditor->text();
```

Comment:
I've used `selectedText().length() > 0` instead of `hasSelectedText()` after looking at https://github.com/qgis/QGIS/commit/c3d9f10cbe10dafb928d5e6416af832c94c9fc37 and https://web.archive.org/web/20230324055015/https://issues.qgis.org/issues/12193. Maybe it is no longer needed now...